sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala

```diff
@@ -137,6 +137,20 @@ class ColumnExpressionSuite extends QueryTest with SharedSparkSession {
     assert(newCol.expr.asInstanceOf[NamedExpression].metadata.getString("key") === "value")
   }
 
+  test("as propagates metadata from nested column") {
```

Comment:
Can you add jira number as test name prefix? I.e. `SPARK-34805: ...`

Reply:
Yes, done.